### PR TITLE
[ADD] eCommerce: checkout process

### PR DESCRIPTION
**Pull request by toaa-odoo**
Adding new page on checkout process for eCommerce. Forward to master.

taskid: 3110554

### CODE REVIEW

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+--> Customize` from any checkout page, and enable :guilabel:`Show B2B fields`.
+
+Request extra info (extra step)
+-------------------------------
+
+You can request :guilabel:`Extra Info` from the customer by adding an **extra step** between the
+:guilabel:`Address` and :guilabel:`Confirm Order` steps. The :guilabel:`Extra Info` step is an
+online form linked to the quotation or sales order of the customer. The information added in the
+extra step is available on the quotation or sales order of the customer from the back end, in the
+**Sales** app.
```
Comment (Donapi) [inline content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:55]:
```suggestion
You can request :guilabel:`Extra Info` from the customer by adding an **extra step** between the
:guilabel:`Address` and :guilabel:`Confirm Order` steps. To do so, go to :menuselection:`Edit
--> Customize` from any checkout page, and enable : :guilabel:`Extra Step Option`
The :guilabel:`Extra Info` step is an online form linked to the quotation or sales order of the customer. The information added in the extra step is available on the quotation or sales order of the customer from the back end, in the **Sales** app.
```

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+When enabled, you can remove, add, and modify fields of the form by clicking on :guilabel:`Edit` in
+the top-right corner, and then clicking on any of the form's fields. At the bottom of the
+:guilabel:`Customize` menu, under the :guilabel:`Field` section, are all the customization options
+for the fields, as well as the :guilabel:`+ Field` button to add additional fields.
+
+.. image:: checkout/checkout-form.png
+   :align: center
+   :alt: Online form customization
```
Comment (Donapi) [inline content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:64]: I would personally prefer to show what's on screen than the customization part
image

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+Confirm Order
+-------------
+
+Terms and Conditions
```
Comment (Donapi) [inline content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:69]:
```suggestion
Terms and conditions
```

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+When enabled, you can remove, add, and modify fields of the form by clicking on :guilabel:`Edit` in
+the top-right corner, and then clicking on any of the form's fields. At the bottom of the
+:guilabel:`Customize` menu, under the :guilabel:`Field` section, are all the customization options
+for the fields, as well as the :guilabel:`+ Field` button to add additional fields.
+
+.. image:: checkout/checkout-form.png
+   :align: center
+   :alt: Online form customization
+
+Confirm Order
```
Comment (Donapi) [inline content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:66]:
```suggestion
Confirm order
```

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+~~~~~~~~~~~~~~~~~~~~
+
+You can ask customers to agree to the :guilabel:`Terms & Conditions` in order to confirm their order
+by enabling :guilabel:`Accept Terms & Conditions` under :menuselection:`Edit --> Customize` on any
+checkout page. The :guilabel:`Terms & Conditions` prompt is found on the :guilabel:`Confirm Order`
+step.
+
+.. image:: checkout/checkout-terms.png
+   :align: center
+   :alt: Terms and conditions
```
Comment (Donapi) [inline content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:79]: I would resize the image that looks very big.

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+.. todo:: add section on suggested accessories when that PR is published
+.. todo:: add ref to discounts when published
+
+Add content
+===========
+
+You can use **building blocks** to add content at any step of the checkout process, including the
+**order confirmation** page. To do so, from any **checkout page**, go to :menuselection:`Edit -->
```
Comment (Donapi) [inline content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:88]:
```suggestion
You can use **building blocks** to add content at any step of the checkout process. To do so, from any **checkout page**, go to :menuselection:`Edit -->
```

Comment (Donapi) [reply content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:88]: I would not mention "incuding the order confirmation page" as you state just before "at any step.

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+each step.
+
+Express checkout
+================
+
+You can enable a :guilabel:`Buy Now` button on products' pages which instantly takes the customer to
+the :guilabel:`Confirm Order` checkout page, instead of adding the product to the cart. To do so, go
+to :menuselection:`Website app --> Configuration --> Settings --> Shop - Checkout Process section`
+and tick :guilabel:`Buy Now`. Alternatively, the :guilabel:`Buy Now` button can also be enabled from
+any product's page by going :menuselection:`Edit --> Customize`.
```
Comment (Donapi) [inline content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:99]:
```suggestion
any product's page by going :menuselection:`Edit --> Customize --> Cart`.
```

**Review by Donapi — APPROVED** [review]
Great page, I enjoyed discovering it :) Just a few comments. 
I already approve the PR to speed up the process. Don't forget to ping @xpl-odoo or @jcs-odoo when it's ready for final review.

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst (hunk 1 above)
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst]: I did add a picture to make it more understandable where that extra page can be found in the process, but not the steps. I don't think it is necessary.

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst (hunk 2 above)
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:67]: Good input

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+Address
+-------
+
+B2B fields
+~~~~~~~~~~
+
+Optional :guilabel:`TIN/VAT` and :guilabel:`Company Name` fields can be added to the
+:guilabel:`Billing Address` form for B2B customers. To add the fields, go to :menuselection:`Edit
+--> Customize` from any checkout page, and enable :guilabel:`Show B2B fields`.
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst]: You have to be on the "Address" step, when you're filing the address. I added that info.

**Review by xpl-odoo — CHANGES_REQUESTED** [review]
Mostly suggestions as usual :)

File: content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+between the :guilabel:`Address` and :guilabel:`Confirm Order` steps. To do so, go to
+:menuselection:`Edit --> Customize` from any checkout page, enable :guilabel:`Extra Step Option`.
+
+.. image:: checkout/checkout-steps.png
+   :align: center
+   :alt: Checkout steps
+
+The :guilabel:`Extra Info` step is an online form linked to the quotation or sales order of the
+customer. The information added in the extra step is available on the quotation or sales order of
+the customer from the back end, in the **Sales** app.
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/checkout.rst:58]: Rewrote it to be clearer

**Review by xpl-odoo — APPROVED** [review]
A couple of minor comments, already approved :)

**Comment by toaa-odoo** [comment]
Do not merge. Part of mega push.

**Comment by toaa-odoo** [comment]
@odoo/doc-review I'm aware of the conflict, I'll fix it before merging. Could you ignore that and have a look at the rest of the PR? Thanks!

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Comment by toaa-odoo** [comment]
@robodoo r+

**Pull request closed without merging**